File: exams/frmFilter.vb
Public Class frmFilter

    Private Sub frmFilter_KeyPress(ByVal sender As Object, ByVal e As System.Windows.Forms.KeyPressEventArgs) Handles Me.KeyPress
    End Sub

    Private Sub frmFilter_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        radNone.Checked = True
        radBottom.Checked = False
        radTop.Checked = False
    End Sub

    Private Sub txtNumber_KeyPress(ByVal sender As Object, ByVal e As System.Windows.Forms.KeyPressEventArgs) Handles txtNumber.KeyPress
        If Asc(e.KeyChar) = 13 Then
            If (Not IsNumeric(txtNumber.Text) And radNone.Checked) Or IsNumeric(txtNumber.Text) And (radBottom.Checked Or radTop.Checked) Then
                radF = radTop.Checked
                radL = radBottom.Checked
                Try
                    rankno = txtNumber.Text
                Catch ex As Exception

                End Try
                Me.Close()
            Else
                failure("Invalid Value For Number Of Students To Filter Considering Your Choice!")
            End If
        End If
    End Sub

    Private Sub Button1_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button1.Click
        If Not (IsNumeric(txtNumber.Text) And radNone.Checked) Or IsNumeric(txtNumber.Text) And (radBottom.Checked Or radTop.Checked) Then
            radF = radTop.Checked
            radL = radBottom.Checked
            Try
                rankno = txtNumber.Text
            Catch ex As Exception

            End Try
            cont = True
            Me.Close()
        Else
            failure("Invalid Value For Number Of Students To Filter Considering Your Choice!")
        End If
    End Sub
End Class

File: exams/frmFilter.cs
using System;
using System.Windows.Forms;
using Microsoft.VisualBasic;
using Microsoft.VisualBasic.CompilerServices;

namespace exams
{
    public partial class frmFilter
    {
        public frmFilter()
        {
            InitializeComponent();
            _Button1.Name = "Button1";
            _txtNumber.Name = "txtNumber";
        }

        private void frmFilter_KeyPress(object sender, KeyPressEventArgs e)
        {
        }

        private void frmFilter_Load(object sender, EventArgs e)
        {
            radNone.Checked = true;
            radBottom.Checked = false;
            radTop.Checked = false;
        }

        private void txtNumber_KeyPress(object sender, KeyPressEventArgs e)
        {
            if (Strings.Asc(e.KeyChar) == 13)
            {
                if (!Information.IsNumeric(txtNumber.Text) & radNone.Checked | Information.IsNumeric(txtNumber.Text) & (radBottom.Checked | radTop.Checked))
                {
                    publicSubsNFunctions.radF = radTop.Checked;
                    publicSubsNFunctions.radL = radBottom.Checked;
                    try
                    {
                        publicSubsNFunctions.rankno = Conversions.ToInteger(txtNumber.Text);
                    }
                    catch (Exception ex)
                    {
                    }

                    Close();
                }
                else
                {
                    publicSubsNFunctions.failure("Invalid Value For Number Of Students To Filter Considering Your Choice!");
                }
            }
        }

        private void Button1_Click(object sender, EventArgs e)
        {
            if (!(Information.IsNumeric(txtNumber.Text) & radNone.Checked) | Information.IsNumeric(txtNumber.Text) & (radBottom.Checked | radTop.Checked))
            {
                publicSubsNFunctions.radF = radTop.Checked;
                publicSubsNFunctions.radL = radBottom.Checked;
                try
                {
                    publicSubsNFunctions.rankno = Conversions.ToInteger(txtNumber.Text);
                }
                catch (Exception ex)
                {
                }

                publicSubsNFunctions.cont = true;
                Close();
            }
            else
            {
                publicSubsNFunctions.failure("Invalid Value For Number Of Students To Filter Considering Your Choice!");
            }
        }
    }
}